client/src/components/TaskModal/LocationInput.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { RiCloseLine } from 'react-icons/ri';
+import { Location } from 'GlobalType';
+
+interface LocationInputProps {
+  locationObject: Location | null;
+  setLocationObject: React.Dispatch<React.SetStateAction<Location | null>>;
+}
+
+const LocationInput = ({ locationObject, setLocationObject }: LocationInputProps) => {
+  const [resultList, setResultList] = useState<any>([]);
+  const [locationInput, setLocationInput] = useState('');
+  const [isLocationInputFocused, setIsLocationInputFocused] = useState<Boolean>(false);
+
+  const OnChangeLocationInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLocationInput(e.target.value);
+    if (e.target.value == '') setResultList([]);
+    else {
+      try {
+        const headers = new Headers({ Authorization: `KakaoAK ${process.env.REACT_APP_REST_API_KEY}` });
+        fetch(`https://dapi.kakao.com/v2/local/search/keyword.json?query=${e.target.value}`, {
+          headers,
+        })
+          .then((res) => res.json())
+          .then((data) => {
+            setResultList(data!.documents);
+          });
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  };
+
+  //하나의 위치 선택
+  const setTagItem = (e: React.MouseEvent<HTMLDivElement>) => {
+    setLocationObject({ location: e.currentTarget!.textContent!, lat: Number(e.currentTarget!.dataset.lat), lng: Number(e.currentTarget!.dataset.lng) });
+  };
+
+  //위치 선택 해제
+  const unSetTagItem = () => {
+    setLocationObject(null);
+    setLocationInput('');
+    setResultList([]);
+  };
+
+  if (locationObject === undefined || locationObject === null)
+    return (
+      <LocationContainer>
+        <InputBar onChange={OnChangeLocationInput} onFocus={(e) => setIsLocationInputFocused(true)} onBlur={(e) => setIsLocationInputFocused(false)} />
+        {isLocationInputFocused ? (
+          <LocationList>
+            {resultList &&
+              resultList.map((item: any) => {
+                return (
```

Comment:
타입에 any를 제거해주세요.
isLocationInputFocused ?  체크 후 resultList &&  resultList 까지 여러번 조건부가 붙습니다.
좀더 깔끔하게 코드 작성하는 법에 대해서 고민해보세요. 복잡하면 컴포넌트로 떼거나 렌더링 함수를 만들어도 됩니다.

Reply:
제 생각에도 `LocationSearchResult` 정도로 분리하면 좋을 것 같습니다
```suggestion
        {isLocationInputFocused && <LocationSearchResult />}
```